tests/test_mock_controller.py

```diff
@@ -717,7 +717,7 @@ async def prepare_louvers(self, louver_ids, target_positions):
         target positions are lined up.
 
         """
-        position = np.full(_NUM_LOUVERS, -1.0, dtype=float)
+        position = np.full(NUM_LOUVERS, -1.0, dtype=float)
```

Comment:
What is special about -1? Is it "do not change this value"?

Reply:
I have checked out the code and read it and I don't understand what this is doing. You go to a lot of trouble to move louver 5 to 100 and the rest to -1, but I don't see anywhere that you check that what you requested happens.

If this was a way of cycling through the louvers (just move one or a few at a time) I suggest a "check_move_louver" method that takes the identify of a lover to move and moves it (or even a list of louvers to move) and verifies that it/they move as expected. Then you can have a test_move_louver command that calls check_move_louver once per louver (or moves several collections, or even moves all possible combinations, depending on how many there are and how long the check method takes).

Reply:
As a matter of fact, you suggested several months that -1 indeed means "do not move". I agree that the code is complex. My intention was to move only a few and to check that they indeed move. I agree that the code can be simplified. I'll consider improving the code once EIE finally has gotten around implementing the louvers control on their side because I am sure that my side will need changes as well.